src/watchers/indexable-term.php

```diff
@@ -0,0 +1,239 @@
+		\add_action( 'created_term', array( $this, 'save_meta' ), PHP_INT_MAX, 3 );
+		\add_action( 'edited_term', array( $this, 'save_meta' ), PHP_INT_MAX, 3 );
+		\add_action( 'deleted_term', array( $this, 'delete_meta' ), PHP_INT_MAX, 3 );
+	}
+
+	/**
+	 * Deletes a term from the index.
+	 *
+	 * @param int    $term_id          The Term ID to delete.
+	 * @param int    $taxonomy_term_id The Taxonomy ID the Term belonged to.
+	 * @param string $taxonomy         The taxonomy name the Term belonged to.
+	 *
+	 * @return void
+	 */
+	public function delete_meta( $term_id, $taxonomy_term_id, $taxonomy ) {
+		try {
+			$indexable = $this->get_indexable( $term_id, $taxonomy, false );
+			$indexable->delete();
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+	}
+
+	/**
+	 * Update the taxonomy meta data on save.
+	 *
+	 * Note: This method is missing functionality to update internal links and incoming links.
+	 *       As this functionality is currently not available for terms, it has not been added in this
+	 *       class yet.
+	 *
+	 * @param int    $term_id          ID of the term to save data for.
+	 * @param int    $taxonomy_term_id The taxonomy_term_id for the term.
+	 * @param string $taxonomy         The taxonomy the term belongs to.
+	 *
+	 * @return void
+	 */
+	public function save_meta( $term_id, $taxonomy_term_id, $taxonomy ) {
+		try {
+			$indexable = $this->get_indexable( $term_id, $taxonomy );
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+
+		$indexable->permalink = $this->get_permalink( $term_id, $taxonomy );
+
+		$term_meta = $this->get_meta_data( $term_id, $taxonomy );
+
+		foreach ( $this->get_meta_lookup() as $meta_key => $indexable_key ) {
+			$indexable->{$indexable_key} = $term_meta[ $meta_key ];
+		}
+
+		$indexable->primary_focus_keyword_score = $this->get_keyword_score( $term_meta['wpseo_focuskw'], $term_meta['wpseo_linkdex'] );
+
+		$indexable->is_robots_noindex = $this->get_noindex_value( $term_meta['wpseo_noindex'] );
+
+		// Not implemented yet.
+		$indexable->is_cornerstone     = 0;
+		$indexable->is_robots_nofollow = 0;
+
+		$indexable->save();
+
+		if ( ! empty( $indexable->id ) ) {
+			$this->save_social_meta( $indexable, $term_meta );
+		}
+	}
+
+	/**
+	 * Retrieves an indexable for a term.
+	 *
+	 * @param int    $term_id     The term the indexable is based upon.
+	 * @param string $taxonomy    The taxonomy the indexable belongs to.
+	 * @param bool   $auto_create Optional. Creates an indexable if it does not exist yet.
+	 *
+	 * @return Indexable The indexable found for the supplied term.
+	 *
+	 * @throws No_Indexable_Found Exception when no indexable could be found for the supplied term.
+	 */
+	protected function get_indexable( $term_id, $taxonomy, $auto_create = true ) {
+		$indexable = Yoast_Model::of_type( 'Indexable' )
+								->where( 'object_id', $term_id )
+								->where( 'object_type', 'term' )
+								->where( 'object_sub_type', $taxonomy )
```

Comment:
There's probably no plugins for this, but technically speaking `taxonomy` is a column with a variable value the same as `post_type`. The combination of `object_id` and `object_type` should be enough to uniquely identify a term. It shouldn't be possible for two terms to have the same `term_id`.